microsoft/PowerToys
# KBM - Keyboard Manager Shortcut Bugs

## Summary of the Pull Request
With this PR, 2 bugs in KBM were fixed.

## PR Checklist

- [ ] **Closes:** #34798 #34510 


## Detailed Description of the Pull Request / Additional comments
KBM bugs are below:

1. KBM shorcuts don't work when the PC is rebooted or Powertoys is restarted.
2. When a shortcut and sending text with a common modifier key are invoked one after the other, KBM does not work.


## Validation Steps Performed

Shortcut mappings for testing are as follows.

Note 1: Shortcut mappings are with AltGr modifier key. Also tested with only Ctrl Left with the Turkish Keyboard Layout or only with Alt Right with the US keyboard layout.
Note 2: Tested with Italian (Italy) keyboard layout.

![Mappings](https://github.com/user-attachments/assets/ed66c69f-658e-4c7a-8b4d-81d760d4d2d0)

Bug 1 Test Steps:

1. Set KBM shortcuts.
2. Close PowerToys.
3. Press AltGR and dont release.
4. Open Powertoys.
5. Press Action keys. Check shortcuts don't work.
6. Release all keys and press again shortcuts. Check this time shortcuts work properly.


Bug 2 Test Steps:

1. Set keyboard layout to Italian (Italy)
2. Set KBM shortcuts as above in screenshot.
3. Press AltGr + I and check X is pressed.
4. Release only I and press " \ " or " ' " . Check related text is sent. ( ■ , ... )
5. All shortcuts above have same AltGr modifier keys. Test with combination of those shortcuts working properly.
